internal/api/tenant.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+import (
+	"github.com/supabase/gotrue/internal/conf"
+	"github.com/supabase/gotrue/internal/storage"
+
+	"github.com/sirupsen/logrus"
+)
+
+// stores per-tenant settings here
+// TODO: need to add tenant-specific rate limiting here
+type Tenant struct {
+	db       *storage.Connection
+	config   *conf.GlobalConfiguration
+	limiters *Limiters
+}
+
+func (t *Tenant) GetConnection() *storage.Connection {
+	return t.db
+}
+
+func NewTenant(config *conf.GlobalConfiguration) (*Tenant, error) {
+	// TODO: close db connection in cleanup
```

Comment:
qq: what needs to be happen before we can close these TODOs (e.g. further refactoring, infra changes)?